sdk/appconfiguration/azure-spring-cloud-appconfiguration-config-web/src/main/java/com/azure/spring/cloud/config/web/pushrefresh/AppConfigurationRefreshEndpoint.java

```diff
@@ -81,15 +85,24 @@ public String refresh(HttpServletRequest request, HttpServletResponse response,
             return HttpStatus.UNAUTHORIZED.getReasonPhrase();
         }
 
+        String syncToken = "";
+
+        JsonNode data = kvReference.findValue(DATA);
+        if (data != null) {
+            JsonNode syncTokenNode = data.findValue(SYNC_TOKEN);
+            if (syncTokenNode != null) {
+                syncToken = syncTokenNode.asText();
+            }
+        }
+
         JsonNode validationResponse = kvReference.findValue(VALIDATION_CODE_KEY);
         if (validationResponse != null) {
             // Validating Web Hook
             return String.format("%s%s\"}", VALIDATION_CODE_FORMAT_START, validationResponse.asText());
         } else {
             if (contextRefresher != null) {
                 if (validation.triggerRefresh()) {
-                    publisher.publishEvent(
-                        new AppConfigurationRefreshEvent(validation.getEndpoint()));
+                    publisher.publishEvent(new AppConfigurationRefreshEvent(validation.getEndpoint(), syncToken));
```

Comment:
is it possible that the `syncToken` is null?

Reply:
It shouldn't but I added a null check before it is used.